src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts

```diff
@@ -179,6 +180,15 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
 			typeof (<any>thing).scheme === 'string';
 	}
 
+	private _setIconForAutomation(options: IShellLaunchConfigResolveOptions, profile: ITerminalProfile): ITerminalProfile {
+		if (options.allowAutomationShell) {
+			const profileClone = deepClone(profile);
+			profileClone.icon = Codicon.tools;
+			return profileClone;
+		}
+		return profile;
+	}
```

Comment:
Can we just set the icon explicitly inside tasks? Automation profiles are also used by debug and it's not that nice with all these icon special cases. Would this work?

```ts
icon: defaultProfile.icon || Codicon.tools,
```

Reply:
I started with that, but the default profile often already has an icon so the `tools` icon will never get used.